plugins/store/kics/main.go

```diff
@@ -171,46 +180,105 @@ func (s *Scanner) createConfig(input *string) (*ScannerConfig, error) {
 	return &config, nil
 }
 
-func (s *Scanner) formatOutput(rawFile, outputFile string) error {
-	file, err := os.Open(rawFile)
+func (s *Scanner) formatOutput(rawOutputDir, outputFile string, reportFormats []string) error {
+	var wg sync.WaitGroup
+	var resultMutex sync.Mutex
+	var result types.Result
+	errCh := make(chan error, len(reportFormats))
+	for _, format := range reportFormats {
```

Comment:
overall, nice idea! however, we are not gaining a lot of performances here by parallelising 2 output formats which are quick ops. imo there is not much gained here compared to having these ops simpler and synchronous

Reply:
I do agree with you 100%, but I don't believe it can be that much simpler.

This would be the synchronous way:

```go
	var result types.Result
	for _, format := range reportFormats {
		switch format {
		case "json":
			var summaryJSON model.Summary
			err := decodeFile(filepath.Join(rawOutputDir, "kics.json"), &summaryJSON)
			if err != nil {
				return fmt.Errorf("failed to decode kics.json: %w", err)
			}

			var misconfigurations []types.Misconfiguration
			for _, q := range summaryJSON.Queries {
				for _, file := range q.Files {
					misconfigurations = append(misconfigurations, types.Misconfiguration{
						Id:          types.Ptr(file.SimilarityID),
						Location:    types.Ptr(file.FileName + "#" + strconv.Itoa(file.Line)),
						Category:    types.Ptr(q.Category + ":" + string(file.IssueType)),
						Message:     types.Ptr(file.KeyActualValue),
						Description: types.Ptr(q.Description),
						Remediation: types.Ptr(file.KeyExpectedValue),
						Severity:    types.Ptr(mapKICSSeverity[q.Severity]),
					})
				}
			}

			result.Vmclarity.Misconfigurations = &misconfigurations
			result.RawJSON = summaryJSON

		case "sarif":
			var summarySarif interface{}
			err := decodeFile(filepath.Join(rawOutputDir, "kics.sarif"), &summarySarif)
			if err != nil {
				return fmt.Errorf("failed to decode kics.sarif: %w", err)
			}

			result.RawSarif = &summarySarif
		
		default:
			return fmt.Errorf("unsupported report format: %s", format)
		}
	}

	if err := result.Export(outputFile); err != nil {
		return fmt.Errorf("failed to save KICS result: %w", err)
	}

	return nil
```

KICS supports these output formats:

```go
	"json"
	"sarif"
	"html"
	"glsast"
	"pdf"
	"sonarqube"
	"cyclonedx"
	"junit"
	"asff"
	"csv"
	"codeclimate"
```
If we need to add even more in the future, this might be a heavier operation.
Will you be okay with keeping this?
I cleaned up the code. <3